src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,36 @@
 package core.basesyntax.service;
 
 import core.basesyntax.dao.StorageDao;
-import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDao storageDao;
+
+    public RegistrationServiceImpl(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        if (user == null || user.getLogin() == null || user.getPassword()
+                == null || user.getAge() == null) {
+            throw new IllegalArgumentException("User or its properties cannot be null");
+        }
+
+        if (user.getLogin().length() < 6 || user.getPassword().length() < 6) {
```

Comment:
add 6 to the constant, and use it in check and in exception message

same for other number